my-app/src/main/java/com/JavaWebServer/app/App.java

```diff
@@ -1,8 +1,18 @@
 package com.JavaWebServer.app;
 
+import java.util.Map;
+import java.util.HashMap;
+
 public class App {
-  public static void main( String[] args) {
-    Server app  = new Server(5000);
+  private static final int PORT_INDEX = 0;
+  private static final String [] KEYS = {"-p","-d"};
+
+  public static void main( String[] args) throws Exception {
+    Map<String,String> parsedArguments = OptionsParser.parse(args, KEYS);
+    int port = Integer.parseInt(parsedArguments.get(KEYS[PORT_INDEX]));
```

Comment:
It seems strange to have a class that can parse your options, but then still have to convert the result after parsing.